src/hffs/fs.py

```diff
@@ -191,26 +227,32 @@ def rm(self, path, recursive=False, maxdepth=None, **kwargs):
         commit_message += f"up to depth {maxdepth} " if maxdepth is not None else ""
         # TODO: use `commit_description` to list all the deleted paths?
         self._api.create_commit(
-            repo_id=repo_id,
-            repo_type=repo_type,
+            repo_id=resolved_path.repo_id,
+            repo_type=resolved_path.repo_type,
             token=self.token,
             operations=operations,
-            revision=self.revision,
+            revision=revision,
```

Comment:
If revision is None, default back to the resolved path one just in case?

```suggestion
            revision=revision or resolved_path.revision,
```